include counters parameter in server_info command

ckeshava opened the pull request:
Fix #4752 (https://github.com/XRPLF/rippled/issues/4752)
This PR attempts to document the usage of the counters parameter in the rippled server_info command.

@mtrippled please let me know if this change accurately reflects the intent of the command

ckeshava committed:
include counters in server_info command

mtrippled commented on content/references/http-websocket-apis/public-api-methods/server-info-methods/server_info.md:
```diff
@@ -40,13 +51,14 @@ An example of the request format:
 [Try it! >](websocket-api-tool.html#server_info)
 
-The request does not take any parameters.
+The request takes in an optional `counters` parameter. It will render metrics pertaining to Job Queue, nodestore and RPC call counters.
```
suggest "metrics including those that pertain to ". Otherwise, the language makes the following seem complete instead of a subset.

mtrippled reviewed (approved):
I left one comment, but other than that, this is good. LGTM :+1:

ckeshava committed:
update the counters feature description

mDuo13 commented on content/references/http-websocket-apis/public-api-methods/server-info-methods/server_info.md:
````diff
@@ -21,6 +21,11 @@ An example of the request format:
+  "command": "server_info",
+  "counters": true
 }
 ```
````
This is not a good way to do multiple examples. People will copy the whole code block and get an error. It's OK to modify the existing example in-place—though, I would change it to `false` unless you're also going to change the example response.

ckeshava replied:
fixed in the latest tip

mDuo13 commented on content/references/http-websocket-apis/public-api-methods/server-info-methods/server_info.md:
````diff
@@ -40,13 +51,14 @@ An example of the request format:
 ```sh
 #Syntax: server_info
````
```suggestion
#Syntax: server_info [counters]
```

mDuo13 commented on content/references/http-websocket-apis/public-api-methods/server-info-methods/server_info.md:
````diff
@@ -40,13 +51,14 @@ An example of the request format:
 ```sh
 #Syntax: server_info
 rippled server_info
+rippled server_info counters # for optional performance statistics
````
Instead of adding a second line, should update the syntax line above, maybe with a note like the ones on [`ledger`'s Commandline syntax](http://xrpl.local/ledger.html)

ckeshava replied:
fixed in the latest tip

mDuo13 commented on content/references/http-websocket-apis/public-api-methods/server-info-methods/server_info.md:
```diff
@@ -129,6 +141,9 @@ The `info` object may have some arrangement of the following fields:
 | `validation_quorum`                 | Number          | Minimum number of trusted validations required to validate a ledger version. Some circumstances may cause the server to require more validations. |
 | `validator_list_expires`            | String          | _(Admin only)_ Either the human readable time, in UTC, when the current validator list expires, the string `unknown` if the server has yet to load a published validator list or the string `never` if the server uses a static validator list. |
+| `counters`            | Object          | This object contains performance metrics pertaining to the RPC Calls (currently executing calls and the completed calls) and the JobQueue. It also contains the details of the nodestore like node_writes, node_reads_total, node_reads_hit, etc|
+| `current_activity`            | Object          | This field contains two arrays for `jobs` and `methods`. |
```
This feels lacking in information. What do the contents of those arrays even represent?

ckeshava replied:
Hmm, I'm not entirely sure.
@mtrippled can you please suggest better docs for this section?

mDuo13 reviewed (changes requested):
There are some issues with this information. Also, the same needs to be done for `server_state` as well.

ckeshava committed:
Update content/references/http-websocket-apis/public-api-methods/server-info-methods/server_info.md

Co-authored-by: Rome Reginelli <mduo13@gmail.com>

ckeshava committed:
update json example to include single instance

ckeshava committed:
comments explaining commandline params

mDuo13 commented on content/references/http-websocket-apis/public-api-methods/server-info-methods/server_info.md:
```diff
@@ -32,21 +33,28 @@ An example of the request format:
     "params": [
         {}
     ]
+},
```
Same thing applies here as the WebSocket version as commented above

ckeshava replied:
@mDuo13 It's been a while since I saw this PR, apologies for that.

The JSON-RPC contains both types of comma-separated examples, is that not what's expected?

ckeshava replied:
Is it better to keep only one of these two examples?

mDuo13 commented on content/references/http-websocket-apis/public-api-methods/server-info-methods/server_info.md:
```diff
@@ -32,21 +33,28 @@ An example of the request format:
 [Try it! >](websocket-api-tool.html#server_info)
 
-The request does not take any parameters.
+The request takes in an optional `counters` parameter. It will render metrics, including those that pertain to Job Queue, nodestore and RPC call counters.
```
Even if there's only one parameter, we should put it in a table so it's harder to miss. Also, for style reasons, we avoid future tense for situations like this.

```suggestion
The request includes the following parameters:

| Field                 | Type    | Required? | Description |
|:----------------------|:--------|:----------|-------------|
| `counters`            | Boolean | No        | If `true`, return metrics about the job queue, ledger store, and API method activity. The default is `false`. |
```

mDuo13 reviewed (commented):
The changes are for the better, but there are still some things I'd like to see changed (including detail on what the `current_activity` contents mean).

intelliot commented:
@ckeshava - please see above

ckeshava committed:
Update content/references/http-websocket-apis/public-api-methods/server-info-methods/server_info.md

Co-authored-by: Rome Reginelli <mduo13@gmail.com>

ckeshava committed:
Update server_info.md

ckeshava commented:
@mDuo13 I have addressed your comments.
@mtrippled can you please help me with the `current_activity` description? I'm falling short of understanding the usage of this field.

mtrippled commented:
> @mDuo13 I have addressed your comments. @mtrippled can you please help me with the `current_activity` description? I'm falling short of understanding the usage of this field.

Those are the items currently being run in the job queue.

amarantha-k commented:
In preparation for the toolchain migration, ported these updates to the redocly_migration branch in PR #2377.
